### ISSUE TITLE
Document that extend-select wins over more specific ignore

### ISSUE THREAD

**Issue by jankatins** [charliermarsh/ruff#1838]
I've the following `example.py` file:

```
print("AA"
      + "BB")
```

My pyproject.toml specifies in the ruff section:

```
[tool.ruff]
extend-select = ["B0", "C4", "ICN", "ISC", "N", "RUF", "SIM"]
ignore = [
    # Explicitly concatenated string should be implicitly concatenated
    "ISC003",
]

```

Running ruff on it, results in this:

```
λ  ruff example.py
example.py:1:7: ISC003 Explicitly concatenated string should be implicitly concatenated
Found 1 error(s).
```

Using 
```
[tool.ruff]
extend-select = ["B0", "C4", "ICN", "ISC", "N", "RUF", "SIM"]
extend-ignore = [
    # Explicitly concatenated string should be implicitly concatenated
    "ISC003",
]
```

works as I wanted it to (no error), but I found that surprising, that the computation rules seems to be

1. Figure out what should be selected by looking at ignore and select settings, letting the more specific rules win
2. Add the extend-* rules and compute again 

I would have expected:
1. Use ignore/select and add the rules from extend-*
2. Compute the selected items by letting the more specific rules win 

Is this intended?

```
λ ruff --version
ruff 0.0.215
```

**Comment by charliermarsh** [comment]
It is intended and it's a little different than the way these work in Flake8. The main reason being that we support extending `pyproject.toml` files, and without these semantics, it can be impossible for child configurations to override their parents, if the parent ignores and selects a specific prefix code.

I typically recommend using `select` over `extend-select` (and same for `ignore`) unless you're using that `pyproject.toml` extension. So in this case, I'd recommend:

```toml
[tool.ruff]
select = ["E", "F", "B0", "C4", "ICN", "ISC", "N", "RUF", "SIM"]
ignore = [
    # Explicitly concatenated string should be implicitly concatenated
    "ISC003",
]
```

(I agree we could improve the docs here.)

**Comment by jankatins** [comment]
My reasoning was that I wanted to get any new defaults enabled and therefore liked the extend-select version. This feels a bit like there needs an overwrite-extend-ignore and so on :-)

In any case: a proposal for a doc extension for the current state is in PR https://github.com/charliermarsh/ruff/pull/1839

**Comment by IliasAarab** [comment]
Trying to run the same initial example as @jankatins but I do not get the initial error anymore:

`example.py`:
```python
print("AA"
      + "BB")
```

```bash
[tool.ruff.lint]
extend-select = ["B0", "C4", "ICN", "ISC", "N", "RUF", "SIM"]
ignore = [
    # Explicitly concatenated string should be implicitly concatenated
    "ISC003",
]
```
```bash
ruff check example.py
All checks passed!
```

Has the behaviour been modified?

**Comment by jankatins** [comment]
I think yes: https://github.com/astral-sh/ruff/pull/2312
